### PR TITLE
Expose `xseq` in `stat_smooth()`

### LINKED ISSUE #5246: Better documentation of `xseq` for `stat_smooth()`
The `xseq` argument is a sort of hidden option for `stat_smooth()` allowing to select where to draw the smooth line.

Would it be relevant to mention it in the documentation and eventually to provide an example?

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix #5246.

Briefly, the `xseq` parameter was already available to users but was undocumented. This PR makes `xseq` a formal argument to `stat_smooth()`, thereby acknowledging its availability.

### CODE REVIEW

File: R/stat-smooth.R [hunk 1]
```diff
@@ -23,6 +23,9 @@
 #' @param fullrange If `TRUE`, the smoothing line gets expanded to the range of the plot,
 #'   potentially beyond the data. This does not extend the line into any additional padding
 #'   created by `expansion`.
+#' @param xseq A numeric vector of values at which the smoother is evaluated.
+#'   Internally, `NULL` defaults to a sequence of `n` equally spaced points for
```
Comment (EvaMaeRey) [inline R/stat-smooth.R:27]: 'When `NULL` (default), xseq is internally evaluated as a sequence of `n` equally

Comment (EvaMaeRey) [reply R/stat-smooth.R:27]: spaced points for continuous data.

**Comment by teunbrand** [comment]
Thanks @EvaMaeRey for the suggestions! I've adapted the wording.

**Review by thomasp85 — APPROVED** [review]
LGTM

**Pull request merged**